app/controllers/answer.rb

```diff
@@ -0,0 +1,11 @@
+post '/answers' do
+  @answer = Answer.new(choice_id: params[:answer])
+  if request.xhr?
+    @answer[:question_id] = @answer.choice.question.id
+    @answer[:user_id] = session[:user_id]
+    @answer[:survey_id] = @answer.question.survey.id
```

Comment:
All of the work in the previous lines, could be done by a method that helps you build params for answers. You could put it in this controller...